### PR TITLE
fix #5

### LINKED ISSUE #5: panic: "Write called after handler finished"
`conn.Write()` causes a panic when the handler function has exited:

```
panic write called after handler finished
```

Can we add a check into

https://github.com/posener/h2conn/blob/13e7df33ed1548e5902d1b32bd6b5906cce194f5/server.go#L72

to make sure our `conn.Write()` returns an error  (instead of causing a panic) when the `conn` is not usable?

### PR DESCRIPTION
**Pull request by jm33-m0**
Added a check to `conn.Write()`, to make sure the connection is usable before writing anything

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -66,12 +66,18 @@ func Accept(w http.ResponseWriter, r *http.Request) (*Conn, error) {
...
 	f http.Flusher
+	r *http.Request
 }
 
 func (w *flushWrite) Write(data []byte) (int, error) {
-	n, err := w.w.Write(data)
-	w.f.Flush()
-	return n, err
+	select {
+	case <-w.r.Context().Done():
```
Comment (posener) [inline server.go]: Insded of `<-w.r.Context().Done()` in a `select` statement, you can check the `e.r.Context().Err()` value, which will make the code simpler and clearer.

File: server.go [hunk 2]
```diff
@@ -66,12 +66,18 @@ func Accept(w http.ResponseWriter, r *http.Request) (*Conn, error) {
 type flushWrite struct {
 	w io.Writer
 	f http.Flusher
+	r *http.Request
```
Comment (posener) [inline server.go]: In the flushWriter you intend to use only the context, so I think it is better to store the context object and not the whole reqeuest.

**Comment by jm33-m0** [comment]
> Another thing that I would like to ask is if you can add a test for this case? Can you reproduce it in a unit test and then see that the fix works?

I would like to, but I an not sure how to do it. I found golang's http2 throws  a panic `write called after handler finished` if I try to use `conn.Write()` when the HTTP handler is done, so I thought adding a check is necessary.

**Comment by posener** [comment]
See https://github.com/posener/h2conn/issues/5#issuecomment-589512073. I don't think that there is anything to fix here.

**Comment by posener** [comment]
I'm closing, thanks for your understanding!

**Pull request closed without merging**